packages/flutter_tools/gradle/src/main/groovy/app_plugin_loader.groovy

```diff
@@ -0,0 +1,38 @@
+import groovy.json.JsonSlurper
+import org.gradle.api.Plugin
+import org.gradle.api.initialization.Settings
+
+apply plugin: FlutterAppPluginLoaderPlugin
+
+class FlutterAppPluginLoaderPlugin implements Plugin<Settings> {
+    @Override
+    void apply(Settings settings) {
+        def flutterProjectRoot = settings.settingsDir.parentFile
+
+        // If this logic is changed, also change the logic in module_plugin_loader.gradle.
```

Comment:
Might be nice if you blocked off the section of the code that needs to be kept in sync with module_plugin_loader.gradle:

// If any of the following code changes, you must update path/to/file
... code ...
// End of code that must be kept in sync with path/to/file

And you probably want a similar comment to be added in path/to/file pointing back at app_plugin_loader.groovy.